packages/core/cache/src/LayeredCache.js

```diff
@@ -0,0 +1,176 @@
+// @flow strict-local
+
+import type {Cache} from './types';
+import type {Readable} from 'stream';
+
+import {
+  serialize,
+  // prepareForSerialization,
+  registerSerializableClass,
+} from '@parcel/core';
+import {blobToString, streamFromPromise} from '@parcel/utils';
+
+// flowlint-next-line untyped-import:off
+import packageJson from '../package.json';
+
+// writes to the local cache are always synchronized to all workers
+//
+// Writes (batched write-through):
+//  - are immediately written to the local cache and
+//  - in Main thread: are batched/debounced and written to the remote cache
+//  - in Workers: sent to the main thread
+// Reads (cannot be batched)
+//  - read from local, fallback to remote. store result in local cache
+
+// type SerializedLayeredCache = {|
+//   local: Cache,
+//   remote: Cache,
+//   batchSize: number,
+//   handle: Handle,
+// |};
+
+// let i = 0;
+
+export class LayeredCache implements Cache {
+  local: Cache;
+  remote: Cache;
+  batchSize: number;
+  _batch: Array<[string, Buffer | string]> = [];
+  // handle: ?Handle;
+  // farm: ?WorkerFarm;
+
+  // #isWorker: boolean = false;
+
+  constructor(
+    local: Cache,
+    remote: Cache,
+    batchSize: number,
+    // workerFarm: WorkerFarm | Handle,
+  ) {
+    this.remote = remote;
+    this.local = local;
+    this.batchSize = batchSize;
+  }
+
+  // static deserialize(opts: SerializedLayeredCache): LayeredCache {
+  //   let cache = new LayeredCache(farm, opts.local, opts.remote, opts.batchSize);
+  //   cache.#isWorker = true;
+  //   return cache;
+  // }
+
+  // serialize(): SerializedLayeredCache {
+  //   if (!this.handle) {
+  //     this.handle = this.farm.createReverseHandle(
+  //       (fn: string, args: Array<mixed>) => {
+  //         // $FlowFixMe
+  //         return this[fn](...args);
+  //       },
+  //     );
+  //   }
+  //   return {
+  //     local: this.local,
+  //     remote: this.remote,
+  //     batchSize: this.batchSize,
+  //     handle: this.handle,
+  //   };
+  // }
+
+  // async flush(): Promise<void> {
+  //   // let v = i++;
+  //   // console.time('flush' + v);
+  //   let batch = this._batch;
+  //   this._batch = [];
+  //   await this.remote.setBlobs(batch);
+  //   // console.timeEnd('flush' + v);
+  // }
+
+  // async #addToBatch(key: string, value: Buffer | string) {
+  //   // if (this.#isWorker) {
+  //   // } else {
+  //   this._batch.push([key, value]);
+  //   if (this._batch.length > this.batchSize) {
+  //     await this.flush();
+  //   }
+  //   // }
+  // }
```

Comment:
I guess this commented out code isn't needed?

Reply:
This whole file isn't used at the moment (and not exported in `cache/src/index`. Same for `RemoteCache.js`). I had experimented with batching writes to the remote cache.

I've deleted all of these files for now